core/chain_manager.go

```diff
@@ -92,6 +92,7 @@ type ChainManager struct {
 	futureBlocks *BlockCache
 
 	quit chan struct{}
+	wg   sync.WaitGroup
```

Comment:
Either way, the issue doesn't really exist anymore because the blockpool is no longer used. It would still be preferable to fix `ChainManager` so it exits properly.

@Gustav-Simonsson maybe you could strip those commits (changes to blockpool and the chain manager) from the PR. I'll submit a separate PR with the change to core.